api.lisp

```diff
@@ -0,0 +1,77 @@
+;;;; -*- Mode: Lisp; indent-tabs-mode: nil -*-
+
+(in-package :split-sequence)
+
+(declaim (inline list-long-enough-p check-bounds check-tests))
+
+(defun list-long-enough-p (list length)
+  (or (= 0 length)
+      (not (null (nthcdr (1- length) list)))))
+
+(defun check-bounds (sequence start end)
+  (progn
+    (check-type start unsigned-byte "a non-negative integer")
+    (check-type end (or null unsigned-byte) "a non-negative integer or NIL")
+    (typecase sequence
+      (list
+       (when end
+         (unless (list-long-enough-p sequence end)
+           (error "The list is too short: END was ~S but the list is ~S elements long."
+                  end (length sequence)))))
+      (t
+       (let ((length (length sequence)))
+         (unless end (setf end length))
+         (unless (<= start end length)
+           (error "Wrong sequence bounds. start: ~S end: ~S" start end)))))))
```

Comment:
The error message on line 19 here uses upper case for the parameter name (`END`) but here we use lower case.  We should probably be consistent.